spring-cloud-sleuth-autoconfigure/src/test/java/org/springframework/cloud/sleuth/autoconfig/wavefront/WavefrontTracingIntegrationTests.java

```diff
@@ -0,0 +1,392 @@
+
+		/**
+		 * Sleuth would automatically wire this, except there's another impl in the
+		 * classpath.
+		 */
+		@Bean
+		Tracer opentracing(Tracing tracing) {
+			return BraveTracer.create(tracing);
+		}
+
+		@Bean
+		BlockingDeque<SpanRecord> spanRecordQueue() {
+			return new LinkedBlockingDeque<>();
+		}
+
+		@Bean
+		@Primary
+		WavefrontSender wavefrontSender(BlockingDeque<SpanRecord> spanRecordQueue) {
+			return new WavefrontSender() {
+				@Override
+				public String getClientId() {
+					return null;
+				}
+
+				@Override
+				public void flush() {
+
+				}
+
+				@Override
+				public int getFailureCount() {
+					return 0;
+				}
+
+				@Override
+				public void sendDistribution(String name, List<Pair<Double, Integer>> centroids,
+						Set<HistogramGranularity> histogramGranularities, Long timestamp, String source,
+						Map<String, String> tags) {
+
+				}
+
+				@Override
+				public void sendMetric(String name, double value, Long timestamp, String source,
+						Map<String, String> tags) {
+
+				}
+
+				@Override
+				public void sendFormattedMetric(String point) {
+
+				}
+
+				@Override
+				public void sendSpan(String name, long startMillis, long durationMillis, String source, UUID traceId,
+						UUID spanId, List<UUID> parents, List<UUID> followsFrom, List<Pair<String, String>> tags,
+						List<SpanLog> spanLogs) {
+					spanRecordQueue.add(new SpanRecord(name, startMillis, durationMillis, source, traceId, spanId,
+							parents, followsFrom, tags, spanLogs));
+				}
+
+				@Override
+				public void close() {
+
+				}
+			};
+		}
+
+		@Bean
+		ApplicationTags applicationTags(Environment environment) {
+			return createFromProperties(environment);
+		}
+
+		public ApplicationTags createFromProperties(Environment environment) {
+			String application = environment.getProperty("wavefront.application.name");
+			String service = environment.getProperty("wavefront.application.service");
+			service = (StringUtils.hasText(service)) ? service : defaultServiceName(environment);
+			ApplicationTags.Builder builder = new ApplicationTags.Builder(application, service);
+			builder.cluster(environment.getProperty("wavefront.application.cluster"));
+			builder.shard(environment.getProperty("wavefront.application.shard"));
+			return builder.build();
+		}
+
+		private String defaultServiceName(Environment environment) {
+			String applicationName = environment.getProperty("spring.application.name");
+			return (StringUtils.hasText(applicationName)) ? applicationName : "unnamed_service";
+		}
+
+	}
+
+	@Controller
+	static class WebMvcController {
+
+		@Autowired
+		Tracer opentracing;
+
+		@Autowired
+		brave.Tracer tracer;
+
+		@RequestMapping("/api/fn/{id}")
+		public ResponseEntity<String> fn(@PathVariable("id") String id) {
+			return new ResponseEntity<>(id, HttpStatus.OK);
+		}
+
+		@RequestMapping("/error/{api}")
+		public ResponseEntity<Void> error(@PathVariable("api") String api) {
+			switch (api) {
+			case "brave":
```

Comment:
Is this indented correctly (switch and case are on the same level)?

Reply:
That's a copy from the previous solution. Let me polish it...

Reply:
So it's the autoformatter that does that apparently